testplan/common/utils/match.py

```diff
@@ -57,6 +58,203 @@ def match_regexps_in_file(
     return all(extracts_status), extracted_values, unmatched
 
 
+class LogFilePosition:
+    """
+    Class for managing the log file positions.
+    """
+
+    def __init__(self, inode: int, position: int):
+        """
+        :param inode: File inode of a log file.
+        :type inode: ``int``
+        :param position: Current position of a log file.
+        :type position: ``int``
+        """
+        self.inode = inode
+        self.position = position
+
+
+class RotationStrategy(object, metaclass=ABCMeta):
+    """
+    Base rotation strategy with abstract methods to manage regex
+    matching in log files. Implement all abstract methods if you
+    need to subclass this class.
+    """
+
+    @abstractmethod
+    def init_strategy(self, log_path: str) -> LogFilePosition:
+        """
+        Returns the initial log file position for the log_path.
+        """
+        raise NotImplementedError
+
+    @abstractmethod
+    def open_logfile_in_position(
+        self, position: LogFilePosition, mode: str
+    ) -> TextIO:
+        """
+        Implementation for opening and returning log handles with the
+        given file position.
+        """
+        raise NotImplementedError
+
+    @abstractmethod
+    def get_position(self, file_handle: TextIO) -> LogFilePosition:
+        """Returns the position for the given file handle."""
+        raise NotImplementedError
+
+    @abstractmethod
+    def get_next_file(
+        self, file_handle: TextIO, mode: Optional[str] = None
+    ) -> Tuple[TextIO, bool]:
+        """Returns the next file handle and a truthy value whether a new file was returned."""
+        raise NotImplementedError
+
+    @abstractmethod
+    def seek_eof(self) -> LogFilePosition:
+        """Returns the end file position of the current log file."""
+        raise NotImplementedError
+
+    @abstractmethod
+    def seek(self, mark: Optional[LogFilePosition] = None) -> LogFilePosition:
```

Comment:
No need for seek, as if mark is None one can just use get_start_of_file_position, if there is a mark this would always return that